contracts/src/BeefyClient.sol

```diff
@@ -323,43 +283,50 @@ contract BeefyClient {
      * @param leafProof an MMR leaf proof
      * @param leafProofOrder a bitfield describing the order of each item (left vs right)
      */
-    function submitFinalWithHandover(
+    function submitFinal(
         Commitment calldata commitment,
         uint256[] calldata bitfield,
         ValidatorProof[] calldata proofs,
         MMRLeaf calldata leaf,
         bytes32[] calldata leafProof,
         uint256 leafProofOrder
-    ) public {
+    ) external {
         (bytes32 commitmentHash, bytes32 ticketID) = validate(commitment, bitfield);
 
-        Ticket storage ticket = tickets[ticketID];
-
-        if (commitment.validatorSetID != nextValidatorSet.id) {
+        bool is_next_session = false;
+        ValidatorSet memory vset = currentValidatorSet;
```

Comment:
See my above comment, we can save gas costs by not loading `currentValidatorSet` unnecessarily.